### PR TITLE
Linda data transfer1

### PR DESCRIPTION
**Pull request by lindaxiang**

### CODE REVIEW

File: _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+breadcrumb: true
+author: junjun_zhang, linda_xiang
+date: 2016-12-12
+categories: linda_xiang
+tags:
+   - GitHub
+   - data transfer
+   - job scheduling
+teaser:
+   info: We use JSON files to represent data transfer jobs. Each JSON file contains necessary information for a data transfer job. JSON files are pre-generated and checked into a GitHub repository, which is structured with a few directories each representing a job state. Each data transfer worker clones the GitHub repository and picks up a queued job by moving the corresponding JSON file from 'queued-jobs' to 'downloading-jobs' directory. It then commits the change and pushes back to GitHub central repository. If more than one worker picked up the same job, a Git merge conflict will occur which elegantly avoids double scheduling. As the transfer job progresses to next steps, the worker will add log to the job JSON file and move it from one directory to the next, every file change and movement will be committed and pushed to GitHub.
```
Comment (btiernay) [inline _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: You might want to abstract this a bit as it seems pretty nuts and bolts. I think focusing on intent in the abstract would help the user have an appreciation for what's to come.

File: _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+### Generate job JSONs
+
+Each job transfers data for a particular GNOS Analysis Object consisting
+of a group of files from a GNOS repository. Job JSON files containing
+information about files to be transfered are generated using metadata
+retrieved from GNOS repositories. Specially designed convention is
+followed to name each job JSON file, which includes multiple parts to
+ensure uniqueness and being informative. For example,
+001a5fa1-dcc8-43e6-8815-fac34eb8a3c9.RECA-EU.C0015.C0015T.WGS-BWA-Tumor.json, it follows this pattern:
```
Comment (btiernay) [inline _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: Might want to put some of these in backticks to better highlight it on the page.

File: _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+---
+
+## The task
+
+The Pan-Cancer Analysis of Whole Genomes
+([PCAWG](https://dcc.icgc.org/pcawg)) project is a collaborative effort by
+the International Cancer Genome Consortium (ICGC) to identify common
+mutation pattens from over 2,800 cancer whole genomes with an emphasize in
+non-coding regions. Large amount of data were generated and stored in
+multiple GNOS repositories, core dataset includes:
```
Comment (btiernay) [inline _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: I would hyperlink to GNOS. Many readers may not know the term.

File: _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md (hunk 1 above)
Comment (junjun-zhang) [reply _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: Was thinking exactly the same. But didn't get much time to do that, PM wants this out asap. Will try.

File: _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md (hunk 2 above)
Comment (junjun-zhang) [reply _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: yes, good idea

File: _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md (hunk 3 above)
Comment (junjun-zhang) [reply _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: I was actually looking around for an appropriate link to GNOS, but not sure which one is the correct one to use. Will try

Comment (junjun-zhang) [reply _posts/linda_xiang/2016-12-07-Cloud-data-transfer.md]: This one might be the right one although it's not under 'Annai' (http://annaisystems.com/):  https://annaisystems.zendesk.com/hc/en-us/sections/200575407-GNOS-and-GeneTorrent

**Review by btiernay — APPROVED** [review]

**Pull request merged**